### PR TITLE
(Update/Module) Fix api builder

### PR DESCRIPTION
**Pull request by pangolp**
In this PR, modify the following:

1. Updated the files to be compatible, with the latest versions of the emulator.
2. Fixed some files, which had tabs instead of spaces.
3. The extra_flag option was added, to avoid the error, of the npc that have gossip, but do not have it assigned in the DB.

I will test the game ingame, because the cmake, it worked without problems and then I will upload a screenshot.

### CODE REVIEW

**Review by Helias — APPROVED** [review]

**Comment by pangolp** [comment]
I upload the images inside the game.
While I wait for the result of the tests.

WoWScrnShot_061522_161736

WoWScrnShot_061522_161822

WoWScrnShot_061522_161830

WoWScrnShot_061522_161917

WoWScrnShot_061522_162014

WoWScrnShot_061522_163157

**Comment by Helias** [comment]
```
The DBErrors.log file contains startup errors:
Script named 'CreatureScript_Professions' is not assigned in the database.
Please solve the startup errors listed above!
```

**Comment by pangolp** [comment]
Let's try something to see if it works. Because I am noticing, that the directory hierarchy inside the SQL is not correct. Let's see if that fixes it.

**Comment by pangolp** [comment]
@Helias Please. If you can, run the tests, and see if this change works well this time.

**Comment by Helias** [comment]
same issue

**Comment by pangolp** [comment]
> same issue

I don't really understand the cause, because in the SQL, it clearly shows, that the script is assigned. Also, I did it in the INSERT, and in the UPDATE. I will find out, what can be the reason, but I really do not understand it.

**Comment by Helias** [comment]
I don't know, there is something missing in the db for sure

**Comment by pangolp** [comment]
> I don't know, there is something missing in the db for sure

I will try to rewrite the SQL script, taking it to the last structure of the emulator, to see if with that, it stops complaining, because the truth is that if not, I don't understand. haha.

**Comment by pangolp** [comment]
If it doesn't come to that, I really don't know what it could be. When you can, run the tests please @Helias. Thank you.

File: src/ProfessionNPC.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #include "ScriptMgr.h"
 #include "Player.h"
 #include "ScriptedGossip.h"
+#include "loader.h"
```
Comment (Winfidonarleyan) [inline src/ProfessionNPC.cpp:4]: No need this line

File: src/loader.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+void AddSC_Professions();
```
Comment (Winfidonarleyan) [inline src/loader.h:2]: This file need .cpp

File: src/ProfessionNPC.cpp (hunk 1 above)
Comment (pangolp) [reply src/ProfessionNPC.cpp:4]: I used to think it was necessary. In fact other modules gave me an error for not using the include. But well, if it is not necessary then I remove it. Thanks for the information.

**Comment by Helias** [comment]
You should make automatically the import of your sql file in this repository to the AzerothCore of the pipeline.
Is the sql file of the module automatically imported through the core when you start it?

image

**Comment by pangolp** [comment]
> You should make automatically the import of your sql file in this repository to the AzerothCore of the pipeline.
> Is the sql file of the module automatically imported through the core when you start it?

I do not have the automatic updates. However, using the db_assembler, I generate the files, use them and have no problems. That's why I don't understand why the test doesn't pass. It's weird, but I think it's something in the test.

**Comment by Helias** [comment]
@Winfidonarleyan @IntelligentQuantum any clue?

**Comment by Winfidonarleyan** [comment]
Need normal dir for sql
https://github.com/azerothcore/mod-cfbg/tree/master/data

**Comment by Helias** [comment]
I think I fixed it! thanks @Winfidonarleyan and @pangolp

**Comment by Winfidonarleyan** [comment]
> I think I fixed it! thanks @Winfidonarleyan and @pangolp 

Yes, fixed, good

**Comment by pangolp** [comment]
What an idiot I was. How could I not see the mistake haha. Thanks to all of you.

**Pull request merged**